docs/chapters/loops/choosing-loops.html

```diff
@@ -0,0 +1,127 @@
+ </div>
+  <input type="hidden" name="check_keywords" value="yes" />
+  <input type="hidden" name="area" value="default" />
+</form>
+          
+        </div>
+    </div>
+  </div>
+
+<div class="container">
+  <div class="row">
+    <div class="col-md-10 col-md-offset-1 content ">
+        <ol class="breadcrumb">
+            <li><a href="index.html">6. Repeating With Loops</a></li>
+          <li class="active">6.7. Choosing Which Loop to Use</li>
+        </ol>
+      
+    
+  <div class="section" id="choosing-which-loop-to-use">
+<h1>6.7. Choosing Which Loop to Use<a class="headerlink" href="#choosing-which-loop-to-use" title="Permalink to this headline">¶</a></h1>
+<p id="index-0">The <code class="docutils literal notranslate"><span class="pre">for</span></code> loop is typically used to iterate through a fixed set of values that can be determined before the loop executes. This is why it is often said that a <code class="docutils literal notranslate"><span class="pre">for</span></code> loop exhibits <strong>definite iteration</strong>.</p>
+<p>On the other hand, the <code class="docutils literal notranslate"><span class="pre">while</span></code> loop is more flexible, as we saw with the example of validating user input. In that case, we could not determine in advance how many times the loop would iterate; it depended entirely on the values provided by the user during program execution. For this reason, a <code class="docutils literal notranslate"><span class="pre">while</span></code> loop is often described as <strong>indefinite iteration</strong>. We expect that <em>eventually</em> the condition controlling the iteration will evaluate to <code class="docutils literal notranslate"><span class="pre">false</span></code> and the iteration will stop. (Unless we have an infinite loop, which is a problem we want to avoid.)</p>
+<p>While we saw that any <code class="docutils literal notranslate"><span class="pre">for</span></code> loop can be written as a <code class="docutils literal notranslate"><span class="pre">while</span></code> loop, by manually creating and updating a loop variable, it is preferable to use a <code class="docutils literal notranslate"><span class="pre">for</span></code> loop when iterating over a collection or iterating a fixed number of times. Manually updating the loop variable in a <code class="docutils literal notranslate"><span class="pre">while</span></code> loop is more work for you, the programmer, and can lead to infinite loops if not handled properly.</p>
+<div class="section" id="check-your-understanding">
+<h2>6.7.1. Check Your Understanding<a class="headerlink" href="#check-your-understanding" title="Permalink to this headline">¶</a></h2>
+<div class="admonition-question admonition">
+<p class="first admonition-title">Question</p>
+<p>You want to write a program that will run until January 1, 2050, redgardless of when it is started. Which loop should you use?</p>
```

Comment:
I feel this could be too hard for them to understand that a while could be checking the current date/time. What do you think about changing it or adding more context?

Reply:
I can see this, since we haven't introduced dates and times. I'll try to think of another conceptual example. Let me know if you have one!

Reply:
@welzie Here is my updated question: "You are asked to program a robot to move tennis balls from one box (Box #1) to another (Box #2), one-by-one. The robot should continue moving balls until Box #1 is empty, and balls may be added while the robot is doing its work. 

   Which type of loop should you use to write the program?"

Thoughts?

Reply:
You mean besides having the word "while" inside the problem asking for which loop to use?  ;-)

I like the question, especially if it is followed by another where the better option is ``for``.

Reply:
Ha! That is a very explicit hint, isn't it? I'll fix that and add another one with `for` as the correct answer.

Reply:
Looks good.